PWGHF/HFC/TableProducer/correlatorD0Hadrons.cxx

```diff
@@ -0,0 +1,491 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file correlatorD0Hadrons.cxx
+/// \brief D0-Hadron correlator task - data-like, MC-reco and MC-kine analyses.
+///
+/// \author Samrangy Sadhu <samrangy.sadhu@cern.ch>, INFN Bari
+/// \author Swapnesh Santosh Khade <swapnesh.santosh.khade@cern.ch>, IIT Indore
+
+#include "Framework/AnalysisTask.h"
+#include "Framework/HistogramRegistry.h"
+#include "PWGHF/DataModel/CandidateReconstructionTables.h"
+#include "PWGHF/DataModel/CandidateSelectionTables.h"
+#include "Common/Core/TrackSelection.h"
+#include "Common/DataModel/TrackSelectionTables.h"
+#include "Framework/runDataProcessing.h"
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+using namespace o2::aod::hf_cand;
+using namespace o2::aod::hf_cand_2prong;
+using namespace o2::aod::hf_correlation_d0_hadron;
+using namespace o2::analysis::hf_cuts_d0_to_pi_k;
+using namespace o2::constants::math;
+
+///
+/// Returns deltaPhi value in range [-pi/2., 3.*pi/2], typically used for correlation studies
+///
+double getDeltaPhi(double phiD, double phiDbar)
+{
+  return RecoDecay::constrainAngle(phiDbar - phiD, -o2::constants::math::PI / 2.);
+}
+
+const int npTBinsMassAndEfficiency = o2::analysis::hf_cuts_d0_to_pi_k::nBinsPt;
+const double efficiencyDmesonDefault[npTBinsMassAndEfficiency] = {};
+auto efficiencyDmeson_v = std::vector<double>{efficiencyDmesonDefault, efficiencyDmesonDefault + npTBinsMassAndEfficiency};
```

Comment:
```suggestion
auto vecEfficiencyDmeson = std::vector<double>{efficiencyDmesonDefault, efficiencyDmesonDefault + npTBinsMassAndEfficiency};
```
Underscores are allowed only in namespace names. (See the [naming rules](https://rawgit.com/AliceO2Group/CodingGuidelines/master/naming_formatting.html?showone=General_Naming_Rules#General_Naming_Rules).)